lib/forms.js

```diff
@@ -13,8 +15,63 @@ var dataTargetImpl = require('./impl/dataTargets');
 var config = require('./config');
 var logger = logfns.getLogger();
 
+function getDbName(uri) {
+  var parsedMongooseUri = mongoUriParser.parse(uri);
+  var dbName = parsedMongooseUri.database;
+  return dbName;
+}
+
+/**
+ * Get a new mongoose connection instance. Will reuse an existing mongoose connection if it's passed in (this connection should be authenticated already and have access to the target db. E.g. admin user). 
+ * Otherwise a new mongoose connection (connection pool) will be created.
+ * @param {string} url the uri of the mongodb
+ * @param {object} mongooseConnection an existing mongoose connection. If exists, it will be resued for the mongoose connection
+ * @param {function} cb the callback function
+ */
+function getMongooseConnection(uri, mongooseConnection, cb) {
+  logger.debug("getting mongoose connection");
+  var mongooseConn;
+  if (mongooseConnection && typeof mongooseConnection.useDb === 'function') {
+    process.nextTick(function() {
+      logger.debug("creating the mongoose connection using the existing connection", {uri: uri});
+      var dbName = getDbName(uri);
+      mongooseConn = mongooseConnection.useDb(dbName);
+      return cb(undefined, mongooseConn);
+    });
+  } else {
+    process.nextTick(function() {
+      logger.debug("creating a new mongoose connection", {uri: uri});
+      mongooseConn = mongoose.createConnection(uri);
+      return cb(undefined, mongooseConn);
+    });
+  }
+}
+
+/**
+ * Get a new mongodb connection instance. Will reuse an existing mongodb connection if it's passed in (this connection should be authenticated already and have access to the target db. E.g. admin user). 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)